src/webview/ConnectionPanel.ts

```diff
@@ -0,0 +1,322 @@
+
+  private readonly _panel: vscode.WebviewPanel;
+  private readonly _extensionUri: vscode.Uri;
+  private _disposables: vscode.Disposable[] = [];
+  private _sidebar: ConnectionsViewProvider | undefined;
+
+  public static getContext(): ExtensionContext {
+    return this._extContext;
+  }
+
+  public static setContext(ec: ExtensionContext) {
+    this._extContext = ec;
+  }
+
+  public static createOrShow(
+    extensionUri: vscode.Uri,
+    sideBar: ConnectionsViewProvider,
+    mode: PanelMode,
+    oldConnectionName?: string,
+  ) {
+    const column = vscode.window.activeTextEditor
+      ? vscode.window.activeTextEditor.viewColumn
+      : undefined;
+
+    // If we already have a panel, show it.
+    if (ConnectionPanel.currentPanel) {
+      ConnectionPanel.currentPanel._panel.reveal(column);
+      return;
+    }
+
+    // Otherwise, create a new panel.
+    let panel: vscode.WebviewPanel;
+    if (mode === PanelMode.Add) {
+      panel = vscode.window.createWebviewPanel(
+        ConnectionPanel.viewType,
+        'Add Connection',
+        column || vscode.ViewColumn.One,
+        getWebviewOptions(extensionUri),
+      );
+      ConnectionPanel.viewType = 'addConnection';
+    } else {
+      panel = vscode.window.createWebviewPanel(
+        ConnectionPanel.viewType,
+        'Edit Connection',
+        column || vscode.ViewColumn.One,
+        getWebviewOptions(extensionUri),
+      );
+      ConnectionPanel.viewType = 'editConnection';
+    }
+
+    ConnectionPanel.currentPanel = new ConnectionPanel(panel, extensionUri, mode);
+    ConnectionPanel.currentPanel._sidebar = sideBar;
+
+    let oldConnection =
+      ConnectionManager.connectionManager.getAllConnections()[oldConnectionName as string];
+
+    ConnectionPanel.currentPanel._panel.webview.postMessage({
+      type: 'Connection.InitializeView',
+      connection: oldConnection,
+    });
+  }
+
+  public static revive(panel: vscode.WebviewPanel, extensionUri: vscode.Uri, mode: PanelMode) {
+    ConnectionPanel.currentPanel = new ConnectionPanel(panel, extensionUri, mode);
+  }
+
+  private constructor(panel: vscode.WebviewPanel, extensionUri: vscode.Uri, mode: PanelMode) {
+    this._panel = panel;
+    this._extensionUri = extensionUri;
+
+    // Set the webview's initial html content
+    this._update(mode);
+
+    // Listen for when the panel is disposed
+    // This happens when the user closes the panel or when the panel is closed programmatically
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+
+    // Update the content based on view changes
+    this._panel.onDidChangeViewState(
+      e => {
+        if (this._panel.visible) {
+          this._update(mode);
+        }
+      },
+      null,
+      this._disposables,
+    );
+
+    // Handle messages from the webview
+    this._panel.webview.onDidReceiveMessage(
+      message => {
+        switch (message.type) {
+          case 'Connection.cancel': {
```

Comment:
These message type strings should exist as constants in a separate file, the same way commands are listed as constants in the `commands.ts` file.